File: app/src/main/java/com/jamburger/kitter/adapters/CommentAdapter.java
package com.jamburger.kitter.adapters;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.jamburger.kitter.OtherProfileActivity;
import com.jamburger.kitter.R;
import com.jamburger.kitter.components.Comment;
import com.jamburger.kitter.components.User;

import java.util.List;

public class CommentAdapter extends RecyclerView.Adapter<CommentAdapter.ViewHolder> {
    Context mContext;
    List<Comment> mComments;

    public CommentAdapter(Context mContext, List<Comment> mComments) {
        this.mContext = mContext;
        this.mComments = mComments;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(mContext).inflate(R.layout.comment_item, parent, false);
        return new CommentAdapter.ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        Comment comment = mComments.get(position);
        comment.getPublisher().get().addOnSuccessListener(snapshot -> {
            User user = snapshot.toObject(User.class);
            Glide.with(mContext).load(user.getProfileImageUrl()).into(holder.profileImage);
            holder.username.setText(user.getUsername());
            holder.comment.setText(comment.getText());

            holder.container.setOnClickListener(view -> {
                Intent intent = new Intent(mContext, OtherProfileActivity.class);
                intent.putExtra("userid", user.getId());
                mContext.startActivity(intent);
            });
        });
    }

    @Override
    public int getItemCount() {
        return mComments.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        ImageView profileImage;
        TextView username, comment;
        View container;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            container = itemView.findViewById(R.id.container);
            profileImage = itemView.findViewById(R.id.img_profile);
            username = itemView.findViewById(R.id.txt_username);
            comment = itemView.findViewById(R.id.txt_comment);
        }
    }
}

File: app/src/main/java/com/jamburger/kitter/adapters/CommentAdapter.kt
package com.jamburger.kitter.adapters

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.google.firebase.firestore.DocumentSnapshot
import com.google.firebase.firestore.FirebaseFirestore
import com.jamburger.kitter.R
import com.jamburger.kitter.activities.OtherProfileActivity
import com.jamburger.kitter.components.Comment
import com.jamburger.kitter.components.User
import com.jamburger.kitter.utilities.DateTimeFormatter

class CommentAdapter(private var mContext: Context, private var mComments: List<Comment>) :
    RecyclerView.Adapter<CommentAdapter.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(mContext).inflate(R.layout.adapter_comment, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val comment = mComments[position]
        val publisherReference =
            FirebaseFirestore.getInstance().collection("Users").document(comment.publisherId)
        publisherReference.get().addOnSuccessListener { snapshot: DocumentSnapshot ->
            val user = snapshot.toObject(
                User::class.java
            )
            Glide.with(mContext).load(user!!.profileImageUrl).into(holder.profileImage)
            holder.username.text = user.username
            holder.time.text = DateTimeFormatter.getTimeDifference(comment.commentId, true)
            holder.comment.text = comment.text
            holder.container.setOnClickListener {
                val intent = Intent(mContext, OtherProfileActivity::class.java)
                intent.putExtra("userid", user.id)
                mContext.startActivity(intent)
            }
        }
    }

    override fun getItemCount(): Int {
        return mComments.size
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var profileImage: ImageView = itemView.findViewById(R.id.img_profile)
        var username: TextView = itemView.findViewById(R.id.txt_username)
        var comment: TextView = itemView.findViewById(R.id.txt_comment)
        var time: TextView = itemView.findViewById(R.id.txt_time)
        var container: View = itemView.findViewById(R.id.container)
    }
}
